### PR TITLE
Dataviews: Add: Ability to delete custom views.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Adds the possibility to delete custom views created by the user.

## Testing Instructions
Delete a view and verify it is deleted and the default view becomes active.

### CODE REVIEW

**Comment by oandregal** [comment]
I've done some testing here:

- When I create a new custom view, the filters are reset (I expected their values to be preserved – this also happens in `trunk` so it can be fixed separately).
- When deleting a custom view, I cannot longer interact with the filters. The default view doesn't become active.

https://github.com/WordPress/gutenberg/assets/583546/be668f4b-e402-4e3e-a83c-a1f75b748b6c

File: packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js [hunk 1]
```diff
@@ -39,6 +55,43 @@ function CustomDataViewItem( { dataviewId, isActive } ) {
...
+									deleteEntityRecord(
+										'postType',
+										'wp_dataviews',
+										dataview.id,
+										{
+											force: true,
+										}
+									);
+									if ( isActive ) {
+										history.push( {
```
Comment (ntsekouras) [inline packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js]: Haven't tested yet, but if we delete, we should probably use `history.replace`.

**Comment by jorgefilipecosta** [comment]
Hi @oandregal,

> When I create a new custom view, the filters are reset (I expected their values to be preserved – this also happens in trunk so it can be fixed separately).

During a conversation with @youknowriad, we discussed the need to preserve filter changes when creating a new view. To achieve this, we would require a provider or a place to store the state view that is easily accessible to both the main area and the sidebar. However, we decided against using a provider and unfortunately, we cannot resolve the issue unless we reconsider our decision of not having a provider.

**Comment by youknowriad** [comment]
> When I create a new custom view, the filters are reset (I expected their values to be preserved – this also happens in trunk so it can be fixed separately).

Yeah, for me there's a difference between creating a new view and "forking an existing view". I don't want to introduce a correlation between a button in the right sidebar and what's happening in the frame. If we want to create a new view based on what's happening on the frame, I think we should have a "fork" button in the frame directly. (There's a lot of subtle implications that result from making the frame and the sidebar dependent on each other and it's not a bridge we should cross I think)

File: packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js (hunk 1 above)
Comment (jorgefilipecosta) [reply packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js]: Good point the code was updated to use replace.

**Comment by jorgefilipecosta** [comment]
Hi @oandregal,
> When deleting a custom view, I cannot longer interact with the filters. The default view doesn't become active.

This was a tricky issue and I spent lots of time to fix but apparently, an await was missing. It should be working now.

**Review by oandregal — APPROVED** [review]

File: packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js [hunk 2]
```diff
@@ -43,6 +56,43 @@ function CustomDataViewItem( { dataviewId, isActive } ) {
 			isActive={ isActive }
 			isCustom="true"
 			customViewId={ dataviewId }
+			suffix={
+				<DropdownMenu
```
Comment (ntsekouras) [inline packages/edit-site/src/components/sidebar-dataviews/custom-dataviews-list.js:60]: This results in nested `button` tags, which is invalid HTML. Additionally when we click the DropDown it also triggers the `click` event in the list item.

**Pull request merged**